packages/mdc-checkbox/mdc-checkbox.scss

```diff
@@ -14,333 +14,135 @@
-    @include mdc-checkbox-cover-element;
-    @include mdc-rtl-reflexive-position(
-      left, ($mdc-checkbox-touch-area - $mdc-checkbox-size) / 2, ".mdc-checkbox");
-
-    display: inline-flex;
-    top: ($mdc-checkbox-touch-area - $mdc-checkbox-size) / 2;
-    align-items: center;
-    justify-content: center;
-    box-sizing: border-box;
-    width: $mdc-checkbox-ui-pct;
-    height: $mdc-checkbox-ui-pct;
-    transition:
-      mdc-checkbox-transition-exit(background-color),
-      mdc-checkbox-transition-exit(border-color);
-    border: $mdc-checkbox-border-width solid $mdc-checkbox-border-color;
-    border-radius: 2px;
-    background-color: transparent;
-    pointer-events: none;
-    will-change: background-color, border-color;
-
-    @include mdc-theme-dark(".mdc-checkbox") {
-      border-color: $mdc-checkbox-border-color-dark;
-    }
-
-    // The frame's ::before element is used as a focus indicator for the checkbox
-    &::before {
-      @include mdc-checkbox-cover-element;
-      @include mdc-theme-prop(background, primary, $edgeOptOut: true);
-
-      width: 100%;
-      height: 100%;
-      transform: scale(0, 0);
-      transition: mdc-checkbox-transition-exit(opacity), mdc-checkbox-transition-exit(transform);
-      border-radius: 50%;
-      opacity: 0;
-      pointer-events: none;
-      content: "";
-      will-change: opacity, transform;
-    }
-  }
-
-  &__native-control {
-    position: absolute;
-    top: 0;
-    left: 0;
-    width: 100%;
-    height: 100%;
-    margin: 0;
-    padding: 0;
-    opacity: 0;
-    cursor: inherit;
-  }
-
-  &__checkmark {
-    @include mdc-checkbox-cover-element;
+  @include mdc-checkbox-ink-color($mdc-checkbox-mark-color);
 
-    width: 100%;
-    transition: mdc-checkbox-transition-exit(opacity, 0ms, $mdc-checkbox-transition-duration * 2);
-    fill: $mdc-checkbox-mark-color;
-    opacity: 0;
-
-    &__path {
-      transition:
-        mdc-checkbox-transition-exit(
-          stroke-dashoffset,
-          0ms,
-          $mdc-checkbox-transition-duration * 2
-        );
-      // !important is needed here because a stroke must be set as an attribute on the SVG in order
-      // for line animation to work properly.
-      stroke: $mdc-checkbox-mark-color !important;
-      stroke-width: $mdc-checkbox-mark-stroke-size * 1.3;
-      stroke-dashoffset: $_mdc-checkbox-mark-path-length;
-      stroke-dasharray: $_mdc-checkbox-mark-path-length;
-    }
+  @include mdc-checkbox-background-unfilled-enabled {
+    @include mdc-checkbox-stroke-color($mdc-checkbox-border-color);
   }
 
-  &__mixedmark {
-    width: 100%;
-    height: floor($mdc-checkbox-mark-stroke-size);
-    transform: scaleX(0) rotate(0deg);
-    transition: mdc-checkbox-transition-exit(opacity), mdc-checkbox-transition-exit(transform);
-    background-color: $mdc-checkbox-mark-color;
-    opacity: 0;
+  @include mdc-checkbox-background-unfilled-disabled {
+    @include mdc-checkbox-stroke-color($mdc-checkbox-disabled-color);
```

Comment:
I should add: `mdc-checkbox-filled-stroke-color` is meant for customizing the stroke color of a filled *enabled* checkbox, so using that mixin to set styles for a disabled checkbox doesn't make sense.

Reply:
I was about to say something like this :)

The nice thing about the button mixins is the user doesn't have to think about "enabled" at all, we apply it for them.

Also, we might be able to just use `:enabled` instead of `:not(:disabled)` here - the reason for `:not(:disabled)` on button was to also support its use for `a` elements across browsers.

Reply:
Done. Too bad there's not an `:unchecked` or `:determinate` selector.